File: backend/src/main/java/com/stjoseph/assessmentsystem/model/Job.java
package com.stjoseph.assessmentsystem.model;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "jobs")
public class Job {
    @Id
    private String id;
    
    private String title;
    private String company;
    private String companyDescription;
    private String companyWebsite;
    private String companyLogo;
    private String location;
    private String workMode; // Remote, Hybrid, On-site
    private JobType type;
    private String salary;
    private String salaryMin;
    private String salaryMax;
    private String currency;
    private String description;
    private List<String> requirements;
    private List<String> responsibilities;
    private List<String> benefits;
    private List<String> skillsRequired;
    private String experienceLevel; // Entry, Mid, Senior
    private Integer minExperience;
    private Integer maxExperience;
    private String educationLevel;
    private String industry;
    private String department;
    private String employmentDuration;
    private LocalDateTime applicationDeadline;
    private String postedBy; // Alumni user ID
    private String postedByName;
    private String postedByEmail;
    private String postedByDesignation;
    private String postedByCompany;
    private LocalDateTime postedAt;
    private String applicationUrl;
    private String contactEmail;
    private String contactPhone;
    private JobStatus status;
    private Integer viewCount;
    private Integer applicationCount;
    
    public enum JobType {
        FULL_TIME, PART_TIME, INTERNSHIP, CONTRACT
    }
    
    public enum JobStatus {
        ACTIVE, EXPIRED, DRAFT
    }
    
    public Job() {
        this.postedAt = LocalDateTime.now();
        this.status = JobStatus.ACTIVE;
        this.viewCount = 0;
        this.applicationCount = 0;
    }
    
    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getCompany() { return company; }
    public void setCompany(String company) { this.company = company; }
    
    public String getCompanyDescription() { return companyDescription; }
    public void setCompanyDescription(String companyDescription) { this.companyDescription = companyDescription; }
    
    public String getCompanyWebsite() { return companyWebsite; }
    public void setCompanyWebsite(String companyWebsite) { this.companyWebsite = companyWebsite; }
    
    public String getCompanyLogo() { return companyLogo; }
    public void setCompanyLogo(String companyLogo) { this.companyLogo = companyLogo; }
    
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    
    public String getWorkMode() { return workMode; }
    public void setWorkMode(String workMode) { this.workMode = workMode; }
    
    public JobType getType() { return type; }
    public void setType(JobType type) { this.type = type; }
    
    public String getSalary() { return salary; }
    public void setSalary(String salary) { this.salary = salary; }
    
    public String getSalaryMin() { return salaryMin; }
    public void setSalaryMin(String salaryMin) { this.salaryMin = salaryMin; }
    
    public String getSalaryMax() { return salaryMax; }
    public void setSalaryMax(String salaryMax) { this.salaryMax = salaryMax; }
    
    public String getCurrency() { return currency; }
    public void setCurrency(String currency) { this.currency = currency; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public List<String> getRequirements() { return requirements; }
    public void setRequirements(List<String> requirements) { this.requirements = requirements; }
    
    public List<String> getResponsibilities() { return responsibilities; }
    public void setResponsibilities(List<String> responsibilities) { this.responsibilities = responsibilities; }
    
    public List<String> getBenefits() { return benefits; }
    public void setBenefits(List<String> benefits) { this.benefits = benefits; }
    
    public List<String> getSkillsRequired() { return skillsRequired; }
    public void setSkillsRequired(List<String> skillsRequired) { this.skillsRequired = skillsRequired; }
    
    public String getExperienceLevel() { return experienceLevel; }
    public void setExperienceLevel(String experienceLevel) { this.experienceLevel = experienceLevel; }
    
    public Integer getMinExperience() { return minExperience; }
    public void setMinExperience(Integer minExperience) { this.minExperience = minExperience; }
    
    public Integer getMaxExperience() { return maxExperience; }
    public void setMaxExperience(Integer maxExperience) { this.maxExperience = maxExperience; }
    
    public String getEducationLevel() { return educationLevel; }
    public void setEducationLevel(String educationLevel) { this.educationLevel = educationLevel; }
    
    public String getIndustry() { return industry; }
    public void setIndustry(String industry) { this.industry = industry; }
    
    public String getDepartment() { return department; }
    public void setDepartment(String department) { this.department = department; }
    
    public String getEmploymentDuration() { return employmentDuration; }
    public void setEmploymentDuration(String employmentDuration) { this.employmentDuration = employmentDuration; }
    
    public LocalDateTime getApplicationDeadline() { return applicationDeadline; }
    public void setApplicationDeadline(LocalDateTime applicationDeadline) { this.applicationDeadline = applicationDeadline; }
    
    public String getPostedBy() { return postedBy; }
    public void setPostedBy(String postedBy) { this.postedBy = postedBy; }
    
    public String getPostedByName() { return postedByName; }
    public void setPostedByName(String postedByName) { this.postedByName = postedByName; }
    
    public String getPostedByEmail() { return postedByEmail; }
    public void setPostedByEmail(String postedByEmail) { this.postedByEmail = postedByEmail; }
    
    public String getPostedByDesignation() { return postedByDesignation; }
    public void setPostedByDesignation(String postedByDesignation) { this.postedByDesignation = postedByDesignation; }
    
    public String getPostedByCompany() { return postedByCompany; }
    public void setPostedByCompany(String postedByCompany) { this.postedByCompany = postedByCompany; }
    
    public LocalDateTime getPostedAt() { return postedAt; }
    public void setPostedAt(LocalDateTime postedAt) { this.postedAt = postedAt; }
    
    public String getApplicationUrl() { return applicationUrl; }
    public void setApplicationUrl(String applicationUrl) { this.applicationUrl = applicationUrl; }
    
    public String getContactEmail() { return contactEmail; }
    public void setContactEmail(String contactEmail) { this.contactEmail = contactEmail; }
    
    public String getContactPhone() { return contactPhone; }
    public void setContactPhone(String contactPhone) { this.contactPhone = contactPhone; }
    
    public JobStatus getStatus() { return status; }
    public void setStatus(JobStatus status) { this.status = status; }
    
    public Integer getViewCount() { return viewCount; }
    public void setViewCount(Integer viewCount) { this.viewCount = viewCount; }
    
    public Integer getApplicationCount() { return applicationCount; }
    public void setApplicationCount(Integer applicationCount) { this.applicationCount = applicationCount; }
}

File: lib/models/job.dart
class Job {
  final String id;
  final String title;
  final String company;
  final String? companyDescription;
  final String? companyWebsite;
  final String location;
  final String? workMode;
  final JobType type;
  final String? salary;
  final String? salaryMin;
  final String? salaryMax;
  final String? currency;
  final String description;
  final List<String> requirements;
  final List<String>? responsibilities;
  final List<String>? benefits;
  final List<String>? skillsRequired;
  final String? experienceLevel;
  final int? minExperience;
  final int? maxExperience;
  final String? educationLevel;
  final String? industry;
  final String? department;
  final String? employmentDuration;
  final DateTime? applicationDeadline;
  final String postedBy;
  final String postedByName;
  final String postedByEmail;
  final String? postedByDesignation;
  final String? postedByCompany;
  final DateTime postedAt;
  final String? applicationUrl;
  final String? contactEmail;
  final String? contactPhone;
  final JobStatus status;
  final int? viewCount;
  final int? applicationCount;

  const Job({
    required this.id,
    required this.title,
    required this.company,
    this.companyDescription,
    this.companyWebsite,
    required this.location,
    this.workMode,
    required this.type,
    this.salary,
    this.salaryMin,
    this.salaryMax,
    this.currency,
    required this.description,
    required this.requirements,
    this.responsibilities,
    this.benefits,
    this.skillsRequired,
    this.experienceLevel,
    this.minExperience,
    this.maxExperience,
    this.educationLevel,
    this.industry,
    this.department,
    this.employmentDuration,
    this.applicationDeadline,
    required this.postedBy,
    required this.postedByName,
    required this.postedByEmail,
    this.postedByDesignation,
    this.postedByCompany,
    required this.postedAt,
    this.applicationUrl,
    this.contactEmail,
    this.contactPhone,
    required this.status,
    this.viewCount,
    this.applicationCount,
  });

  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      company: json['company'] ?? '',
      companyDescription: json['companyDescription'],
      companyWebsite: json['companyWebsite'],
      location: json['location'] ?? '',
      workMode: json['workMode'],
      type: JobType.values.firstWhere(
        (e) => e.name.toUpperCase() == (json['type'] ?? 'FULL_TIME').toString().toUpperCase(),
        orElse: () => JobType.FULL_TIME,
      ),
      salary: json['salary'],
      salaryMin: json['salaryMin'],
      salaryMax: json['salaryMax'],
      currency: json['currency'],
      description: json['description'] ?? '',
      requirements: json['requirements'] != null ? List<String>.from(json['requirements']) : [],
      responsibilities: json['responsibilities'] != null ? List<String>.from(json['responsibilities']) : null,
      benefits: json['benefits'] != null ? List<String>.from(json['benefits']) : null,
      skillsRequired: json['skillsRequired'] != null ? List<String>.from(json['skillsRequired']) : null,
      experienceLevel: json['experienceLevel'],
      minExperience: json['minExperience'],
      maxExperience: json['maxExperience'],
      educationLevel: json['educationLevel'],
      industry: json['industry'],
      department: json['department'],
      employmentDuration: json['employmentDuration'],
      applicationDeadline: json['applicationDeadline'] != null ? DateTime.parse(json['applicationDeadline']) : null,
      postedBy: json['postedBy'] ?? '',
      postedByName: json['postedByName'] ?? '',
      postedByEmail: json['postedByEmail'] ?? '',
      postedByDesignation: json['postedByDesignation'],
      postedByCompany: json['postedByCompany'],
      postedAt: DateTime.parse(json['postedAt'] ?? DateTime.now().toIso8601String()),
      applicationUrl: json['applicationUrl'],
      contactEmail: json['contactEmail'],
      contactPhone: json['contactPhone'],
      status: JobStatus.values.firstWhere(
        (e) => e.name.toUpperCase() == (json['status'] ?? 'ACTIVE').toString().toUpperCase(),
        orElse: () => JobStatus.ACTIVE,
      ),
      viewCount: json['viewCount'],
      applicationCount: json['applicationCount'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'company': company,
      'companyDescription': companyDescription,
      'companyWebsite': companyWebsite,
      'location': location,
      'workMode': workMode,
      'type': type.name.toUpperCase(),
      'salary': salary,
      'salaryMin': salaryMin,
      'salaryMax': salaryMax,
      'currency': currency,
      'description': description,
      'requirements': requirements,
      'responsibilities': responsibilities,
      'benefits': benefits,
      'skillsRequired': skillsRequired,
      'experienceLevel': experienceLevel,
      'minExperience': minExperience,
      'maxExperience': maxExperience,
      'educationLevel': educationLevel,
      'industry': industry,
      'department': department,
      'employmentDuration': employmentDuration,
      'applicationDeadline': applicationDeadline?.toIso8601String(),
      'postedBy': postedBy,
      'postedByName': postedByName,
      'postedByEmail': postedByEmail,
      'postedByDesignation': postedByDesignation,
      'postedByCompany': postedByCompany,
      'postedAt': postedAt.toIso8601String(),
      'applicationUrl': applicationUrl,
      'contactEmail': contactEmail,
      'contactPhone': contactPhone,
      'status': status.name.toUpperCase(),
      'viewCount': viewCount,
      'applicationCount': applicationCount,
    };
  }
}

enum JobType {
  FULL_TIME,
  PART_TIME,
  INTERNSHIP,
  CONTRACT,
}

enum JobStatus {
  ACTIVE,
  EXPIRED,
  DRAFT,
}

extension JobTypeExtension on JobType {
  String get displayName {
    switch (this) {
      case JobType.FULL_TIME:
        return 'Full-time';
      case JobType.PART_TIME:
        return 'Part-time';
      case JobType.INTERNSHIP:
        return 'Internship';
      case JobType.CONTRACT:
        return 'Contract';
    }
  }
}